### PR TITLE
[DDOS-400] | Update Alamofire dependency

### PR DESCRIPTION
**Pull request by VolodymyrYatskanychItomych**
# Description

- update Alamofire version to 5.10.1
- add Pods and SPM files to gitignore
- add `NSLock` for thread-safe access to `credential`
- use `@unchecked Sendable` mark for `AlamofireSessionRenewer`

# Task

[Task link](https://dashdevs.atlassian.net/browse/DDOS-400)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,6 +19,14 @@ DerivedData
 *.hmap
 *.ipa
 
+# Swift Package Manager
+Package.resolved
```
Comment (yakovshkolnikov) [inline .gitignore]: Don't exclude `Package.resolved` and `Podfile.lock` as they're used to constrain specific version. Also, bring back `Podfile.lock` file and add `Package.resolved` to the tracked files.

File: Package.swift [hunk 2]
```diff
@@ -10,7 +10,7 @@ let package = Package(
         .library(name: "AlamofireSessionRenewer", targets: ["AlamofireSessionRenewer"]),
     ],
     dependencies: [
-        .package(url: "https://github.com/Alamofire/Alamofire.git", .upToNextMajor(from: "5.4.3"))
+        .package(url: "https://github.com/Alamofire/Alamofire.git", .upToNextMajor(from: "5.10.1"))
```
Comment (yakovshkolnikov) [inline Package.swift]: No needs to change that, you just need to remove Package.resolved and open the Xcode and it will download the latest version (as the major version wasn't bumped).

File: Sources/AlamofireSessionRenewer/AlamofireSessionRenewer.swift [hunk 3]
```diff
@@ -13,7 +13,12 @@ public typealias FailureRenewHandler = (Bool) -> Void
 public typealias RenewCredentialHandler = ((@escaping SuccessRenewHandler, @escaping FailureRenewHandler) -> Void)
 
 /// This class is responsible for authentication credentials renewing process
-open class AlamofireSessionRenewer: RequestInterceptor {
+open class AlamofireSessionRenewer: RequestInterceptor, @unchecked Sendable {
```
Comment (yakovshkolnikov) [inline Sources/AlamofireSessionRenewer/AlamofireSessionRenewer.swift]: Why these changes are needed? And why do you use lock for credentials?

**Review by yakovshkolnikov — CHANGES_REQUESTED** [review]

File: Sources/AlamofireSessionRenewer/AlamofireSessionRenewer.swift (hunk 3 above)
Comment (VolodymyrYatskanychItomych) [reply Sources/AlamofireSessionRenewer/AlamofireSessionRenewer.swift]: > Non-final class 'AlamofireSessionRenewer' cannot conform to 'Sendable'; use '@unchecked Sendable'; this is an error in the Swift 6 language mode

> Stored property '_credential' of 'Sendable'-conforming class 'AlamofireSessionRenewer' is mutable; this is an error in the Swift 6 language mode

We got these errors, so to fix the first one I added `@unchecked Sendable` mark.

And then since the `credentials` property is public, I made it thread safe

Comment (yakovshkolnikov) [reply Sources/AlamofireSessionRenewer/AlamofireSessionRenewer.swift]: But we haven't updated to the swift 6 yet. If these are just a warnings let's keep them and fix in the scope of adding swift concurrency support.

**Review by yakovshkolnikov — APPROVED** [review]
